core/src/main/scala/kafka/server/ControllerServer.scala

```diff
@@ -153,7 +153,7 @@ class ControllerServer(
       val apiVersionManager = new SimpleApiVersionManager(
         ListenerType.CONTROLLER,
         config.unstableApiVersionsEnabled,
-        () => featuresPublisher.features()
```

Comment:
If we decide to fetch features from cache rather than `featuresPublisher`, we can remove whole `featuresPublisher` directly.

Reply:
Removed it. Thanks.